src/validate_i18n.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import re
+
+from flask import request
+from werkzeug.exceptions import abort
+from functools import wraps
+
+DEFAULT_YEAR = '2019'
+DEFAULT_LANG = 'en'
+SUPPORTED_YEARS = {
+    '2019': ('en', 'ja')
+}
+
+
+def validate_i18n(func):
+    @wraps(func)
+    def decorated_function(*args, **kwargs):
+        lang = kwargs.get('lang')
+        year = kwargs.get('year')
+
+        updated_kwargs = validate(lang, year)
+
+        kwargs.update(updated_kwargs)
+
+        return func(*args, **kwargs)
+
+    return decorated_function
+
+
+def validate(lang, year):
+    if year is None:
+        logging.debug('Defaulting the year to: %s' % year)
+        year = DEFAULT_YEAR
+
+    if year not in SUPPORTED_YEARS:
+        logging.debug('Unsupported year requested: %s' % year)
+        abort(404)
+
+    supported_langs = SUPPORTED_YEARS[year]
+    logging.debug('Languages supported for %s: %s.' % (year, supported_langs))
+
+    # If an unsupported language code is passed in, abort.
+    # TODO: Should we still redirect back to default?
+    if lang is not None and lang not in supported_langs:
+        logging.debug('Unsupported language set: %s.' % lang)
+        abort(404)
```

Comment:
The "unsupported language" error message would be helpful to display to the user.

Reply:
How would it be best to surface that error message? At the moment its a 404, should we have a custom 404 page that would show the user that they entered an unsupported language? This condition should only be hit if the user explicitly enters an incorrect language code into the URL. If it's a part of the `Accept-Language` header, it will default instead.

Reply:
Yeah custom error pages would be great. For reference see https://github.com/HTTPArchive/httparchive.org/blob/86d844ebcaa11d85a9197b1d476b1cf178c513df/main.py#L235-L264

Reply:
Ok cool, added TODOs in both places that we call abort(), and I'll address it in a follow-up PR.